datastore/storagebackend/postgresql/gettsattrcombos.go

```diff
@@ -0,0 +1,177 @@
+package postgresql
+
+import (
+	"datastore/datastore"
+	"fmt"
+	"strings"
+
+	_ "github.com/lib/pq"
+)
+
+var (
+	supAttrs map[string]struct{} // supported attributes
+)
+
+func init() {
+
+	supAttrs = map[string]struct{}{}
+
+	for _, sa := range []string{
+		"version",
+		"type",
+		"title",
+		"summary",
+		"keywords",
+		"keywords_vocabulary",
+		"license",
+		"conventions",
+		"naming_authority",
+		"creator_type",
+		"creator_name",
+		"creator_email",
+		"creator_url",
+		"institution",
+		"project",
+		"source",
+		"platform",
+		"platform_vocabulary",
+		"standard_name",
+		"unit",
+		"instrument",
+		"instrument_vocabulary",
+		// TODO: support links
+	} {
+		supAttrs[sa] = struct{}{}
+	}
+}
+
+// getTSAttrCols ... (TODO: add documentation)
+func getTSAttrCols(attrs []string) ([]string, error) {
+	seen := map[string]struct{}{}
+
+	cols := []string{}
+
+	for _, attr := range attrs {
+		attr0 := strings.ToLower(strings.TrimSpace(attr))
+		if _, found := seen[attr0]; found {
+			return nil, fmt.Errorf("attribute >%s< specified more than once", attr0)
+		}
+
+		if _, found := supAttrs[attr0]; found {
+			cols = append(cols, attr0)
+			seen[attr0] = struct{}{}
+		} else {
+			return nil, fmt.Errorf("unsupported attribute: >%s<", attr0)
+		}
+	}
+
+	return cols, nil
+}
+
+// getTSMdata ... (TODO: add documentation)
+func getTSMData(m map[string]interface{}) (*datastore.TSMetadata, error) {
+
+	tsMData := datastore.TSMetadata{}
+
+	for k, v := range m {
+		switch k {
+		case "version":
+			tsMData.Version = v.(string)
+		case "type":
+			tsMData.Type = v.(string)
+		case "title":
+			tsMData.Title = v.(string)
+		case "summary":
+			tsMData.Summary = v.(string)
+		case "keywords":
+			tsMData.Keywords = v.(string)
+		case "keywords_vocabulary":
+			tsMData.KeywordsVocabulary = v.(string)
+		case "license":
+			tsMData.License = v.(string)
+		case "conventions":
+			tsMData.Conventions = v.(string)
+		case "naming_authority":
+			tsMData.NamingAuthority = v.(string)
+		case "creator_type":
+			tsMData.CreatorType = v.(string)
+		case "creator_name":
+			tsMData.CreatorName = v.(string)
+		case "creator_email":
+			tsMData.CreatorEmail = v.(string)
+		case "creator_url":
+			tsMData.CreatorUrl = v.(string)
+		case "institution":
+			tsMData.Institution = v.(string)
+		case "project":
+			tsMData.Project = v.(string)
+		case "source":
+			tsMData.Source = v.(string)
+		case "platform":
+			tsMData.Platform = v.(string)
+		case "platform_vocabulary":
+			tsMData.PlatformVocabulary = v.(string)
+		case "standard_name":
+			tsMData.StandardName = v.(string)
+		case "unit":
+			tsMData.Unit = v.(string)
+		case "instrument":
+			tsMData.Instrument = v.(string)
```

Comment:
Again, good point. Will do.

Reply:
Hm ... I think a function like `func (x *TSMetadata) SetValue(key string, val interface{}) {...}` would be needed to avoid writing all of the above lines explicitly, but I don't think that is available. Hm ...

Reply:
At least such a function doesn't seem to be generated by protoc, but maybe use the general reflection mechanism in golang?

Reply:
Yup, the reflect package in golang seems to do the trick.